tests/test_rotations.py

```diff
@@ -0,0 +1,338 @@
+    # an identity rotation.
+    (rot * rot.inv()).get_rotation_vector()
+
+
+def test_relative_roation():
+    """Test the relative rotation between two rotations."""
+
+    # Set default values for global parameters.
+    mpy.set_default_values()
+
+    rot1 = Rotation([1, 2, 3], 2)
+    rot2 = Rotation([0.1, -0.2, 2], np.pi / 5)
+
+    rot21 = get_relative_rotation(rot1, rot2)
+
+    assert rot2 == rot21 * rot1
+
+
+def test_rotation_vector():
+    """Test if the rotation vector functions give a correct result."""
+
+    # Calculate rotation vector and quaternion.
+    axis = np.array([1.36568, -2.96784, 3.23346878])
+    angle = 0.7189467
+    rotation_vector = angle * axis / np.linalg.norm(axis)
+    q = np.zeros(4)
+    q[0] = np.cos(angle / 2)
+    q[1:] = np.sin(angle / 2) * axis / np.linalg.norm(axis)
+
+    # Check that the rotation object from the quaternion and rotation
+    # vector are equal.
+    rotation_from_vec = Rotation.from_rotation_vector(rotation_vector)
+    assert Rotation.from_quaternion(q) == rotation_from_vec
+    assert Rotation(axis, angle) == rotation_from_vec
+
+    # Check that the same rotation vector is returned after being converted
+    # to a quaternion.
+    np.testing.assert_array_less(
+        np.linalg.norm(rotation_vector - rotation_from_vec.get_rotation_vector()),
+        mpy.eps_quaternion,
+        "test_rotation_vector",
+    )
+
+
+def test_rotation_operator_overload():
+    """Test if the operator overloading gives a correct result."""
+
+    # Calculate rotation and vector.
+    axis = np.array([1.36568, -2.96784, 3.23346878])
+    angle = 0.7189467
+    rot = Rotation(axis, angle)
+    vector = [2.234234, -4.213234, 6.345234]
+
+    # Check the result of the operator overloading.
+    result_vector = np.dot(rot.get_rotation_matrix(), vector)
+    np.testing.assert_array_less(
+        np.linalg.norm(result_vector - rot * vector),
+        mpy.eps_quaternion,
+        "test_rotation_vector",
+    )
+    np.testing.assert_array_less(
+        np.linalg.norm(result_vector - rot * np.array(vector)),
+        mpy.eps_quaternion,
+        "test_rotation_vector",
+    )
+
+
+def test_rotation_matrix():
+    """Test if the correct quaternions are generated from a rotation matrix."""
+
+    # Do one calculation for each case in
+    # Rotation().from_rotation_matrix().
+    vectors = [
+        [[1, 0, 0], [0, -1, 0]],
+        [[0, 0, 1], [0, 1, 0]],
+        [[-1, 0, 0], [0, 1, 0]],
+        [[0, 1, 0], [0, 0, 1]],
+    ]
+
+    for t1, t2 in vectors:
+        rot = Rotation().from_basis(t1, t2)
+        t1_rot = rot * [1, 0, 0]
+        t2_rot = rot * [0, 1, 0]
+        np.testing.assert_array_less(
+            np.linalg.norm(t1 - t1_rot),
+            mpy.eps_quaternion,
+            "test_rotation_matrix: compare t1",
+        )
+        np.testing.assert_array_less(
+            np.linalg.norm(t2 - t2_rot),
+            mpy.eps_quaternion,
+            "test_rotation_matrix: compare t2",
```

Comment:
General comment, I used to do this in a lot of cases, write a short text of what is compared here. I stoped doing so a while ago, simply because I couldn't see any benefit of doing so. I would be fine with removing those (unless for some reason they are super meaningful) from the testing framework. Does not have to be part of this PR.